nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ListenTCP.java

```diff
@@ -87,122 +96,200 @@
             .displayName("Pool Receive Buffers")
             .description(
-                    "When turned on, the processor uses pre-populated pool of buffers when receiving messages. " +
-                    "This is prepared during initialisation of the processor. " +
-                    "With high value of Max Number of TCP Connections and Receive Buffer Size this strategy might allocate significant amount of memory! " +
-                    "When turned off, the byte buffers will be created on demand and be destroyed after use.")
-            .required(true)
+                    "This property is deprecated and no longer used.")
+            .required(false)
             .defaultValue("True")
             .allowableValues("True", "False")
             .addValidator(StandardValidators.BOOLEAN_VALIDATOR)
             .build();
 
-    @Override
-    protected List<PropertyDescriptor> getAdditionalProperties() {
-        return Arrays.asList(
-                MAX_CONNECTIONS,
-                MAX_RECV_THREAD_POOL_SIZE,
-                POOL_RECV_BUFFERS,
-                SSL_CONTEXT_SERVICE,
-                CLIENT_AUTH
-        );
-    }
+    public static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("Messages received successfully will be sent out this relationship.")
+            .build();
+
+    protected List<PropertyDescriptor> descriptors;
+    protected Set<Relationship> relationships;
+    protected volatile int port;
+    protected volatile BlockingQueue<ByteArrayMessage> events;
+    protected volatile BlockingQueue<ByteArrayMessage> errorEvents;
+    protected volatile EventServer eventServer;
+    protected volatile byte[] messageDemarcatorBytes;
+    protected volatile EventBatcher eventBatcher;
 
     @Override
-    protected Collection<ValidationResult> customValidate(final ValidationContext validationContext) {
-        final List<ValidationResult> results = new ArrayList<>();
+    protected void init(final ProcessorInitializationContext context) {
+        final List<PropertyDescriptor> descriptors = new ArrayList<>();
+        descriptors.add(ListenerProperties.PORT);
+        descriptors.add(ListenerProperties.RECV_BUFFER_SIZE);
+        descriptors.add(ListenerProperties.MAX_MESSAGE_QUEUE_SIZE);
+        // Deprecated
+        descriptors.add(ListenerProperties.MAX_SOCKET_BUFFER_SIZE);
+        descriptors.add(ListenerProperties.CHARSET);
+        descriptors.add(ListenerProperties.MAX_CONNECTIONS);
+        descriptors.add(ListenerProperties.MAX_BATCH_SIZE);
+        descriptors.add(ListenerProperties.MESSAGE_DELIMITER);
+        // Deprecated
+        descriptors.add(MAX_RECV_THREAD_POOL_SIZE);
+        // Deprecated
+        descriptors.add(POOL_RECV_BUFFERS);
+        descriptors.add(ListenerProperties.NETWORK_INTF_NAME);
+        descriptors.add(CLIENT_AUTH);
+        descriptors.add(SSL_CONTEXT_SERVICE);
```

Comment:
The order of these two properties should be reversed to retain the current order.
```suggestion
        descriptors.add(SSL_CONTEXT_SERVICE);
        descriptors.add(CLIENT_AUTH);
```